src/nccl_ofi_sendrecv.cpp

```diff
@@ -2344,11 +2319,10 @@ nccl_net_ofi_sendrecv_device_t::nccl_net_ofi_sendrecv_device_t(nccl_net_ofi_plug
 	this->prov_name = this->info->fabric_attr->prov_name;
 
 	/* Create fabric */
-	ret = fi_fabric(this->info->fabric_attr, &this->fabric, nullptr);
-	if (OFI_UNLIKELY(ret != 0)) {
-		NCCL_OFI_WARN("Couldn't open a fabric provider. RC: %d, ERROR: %s",
-			      ret, fi_strerror(-ret));
-		throw std::runtime_error("SENDRECV device constructor: fi_fabric failed");
+	this->fabric = nccl_ofi_ofiutils_fabric_create(this->info);
+	if (OFI_UNLIKELY(!this->fabric)) {
```

Comment:
Nit: add `OFI_UNLIKELY` to other failure checks for Libfabric resource creation.